Add excludeCredentials to Passkey Creation Process

Paul-Taiwo opened the pull request:
This PR introduces the `excludeCredentials` property to the passkey creation flow. This feature enhances the security and usability of the passkey creation process by preventing duplicate registrations of credentials associated with the same user.

dagnelies commented:
Hi @Paul-Taiwo . Thank you very much for this PR. Although it's good, I slightly hesitate to accept it for various small reasons.

1. I think it's a better practice to just set the user ID in order to avoid multiple passkeys for the same user. It's also simpler than passing lists of credential IDs around. It also has the advantage of overwriting the existing credential (if there is any) which often provides better UX (overwrite instead of error) and may also help overwrite a problematic credential.
2. Less to document, understand, test, ...
3. Allowing multiple accounts, like one for work and a personal one, or for multiple users on the same device, sounds more sensible to me.

Actually, I'm more inclined to revert the random user.id generation (it was a hash of the username before) or add an option to add arbitrary properties.

Paul-Taiwo commented:
Hi @dagnelies

Thank you for your feedback on my earlier PR. I’ve taken your suggestions into account and have made the following updates:

1. Removed `excludeCredentials`
2. Added `customProperties` Option:
To retain flexibility for advanced use cases, I’ve introduced an optional customProperties field. This allows developers to add arbitrary properties to the creationOptions object without altering the default behavior.

dagnelies commented:
Sounds great 👍 

What do you think about adding the custom properties to both the registration and authentication? I think it would be more versatile and intuitive that way. Also, the docs must be updated so that people are aware this even exists.

Besides this nitpicking, it looks good and meaningful. Thanks for the contribution

Paul-Taiwo commented:
Hi @dagnelies,
I’ve added custom properties to the authentication and updated the documentation accordingly. Please let me know if there are any other changes you'd like me to make.
